doc/news/_preparation_next_release.md

```diff
@@ -25,11 +25,26 @@ You can also read the news [on our website](https://www.libelektra.org/news/0.9.
 
 ## Highlights
 
-- <<HIGHLIGHT1>>
+- Important Changes to Keynames
 - <<HIGHLIGHT2>>
 - <<HIGHLIGHT3>>
 
-### <<HIGHLIGHT1>>
+### Important Changes to Keynames
+
+There have been significant changes to Elektra's Keynames.
+The most important change is that you now need a `:` after the namespace.
+So instead of `system/elektra/version` you have to use `system:/elektra/version`.
+
+The second big change is to array elements.
+From now on `keyNew ("/array/#10", KEY_END)` will create a `Key` with name `/array/#_10`, to make arrays more user-friendly while preserving numerical ordering.
+
+The whole internal implementation for `keySetName`, `keyAddName`, etc. has been completely rewritten.
+It should now be more performant since we try to do more work with fewer iterations of the string.
+However, there are no benchmarks (yet), so your mileage may vary.
+
+If you rely on specific behaviour of Elektra's Keynames and have already taken the two changes above into account, please refer to the newly created [Keyname documentation](../keynames) and (easier to read) [Python reference implementation](../keynames/keynames.py).
```

Comment:
Did you look at the C-implementation? Did you actually try and understand it?

Just the fact, that the C-implementation validates the whole key name in reverse makes it much more difficult to understand. You even said in another comment "Function is very long"...

Can you give me one concrete example where the Python implementation is harder to read?

To be clear "easier to read" refers to the fact that in the Python implementation it is much easier to spot what key names are valid and why. With just the C-implementation it is quite difficult to see, if a certain key name is valid or not, without executing the code.

Reply:
I mean the fallback in KDB's bootstrap code. As there are quite many important breaking changes it would be best if you can make a `git diff` and look through.